internal/gcsx/multi_range_downloader_wrapper.go

```diff
@@ -220,6 +232,9 @@ func (mrdWrapper *MultiRangeDownloaderWrapper) Read(ctx context.Context, buf []b
 
 		if e != nil && e != io.EOF {
 			e = fmt.Errorf("Error in Add Call: %w", e)
+			mrdWrapper.muCreateMRD.Lock()
```

Comment:
Couple of things:
1. We shouldn't create mrd on every error. Should be created only when mrd is closed by go-sdk.
2. If there are x callbacks failed on the mrd, then we end up creating mrd x number of times even with the muCreateMRD lock. This is the thing which we wanted to avoid in the first place and didn't take this route.

Lets work with go-sdk in coming up with the right solution.